src/Stratis.Bitcoin.IntegrationTests/Wallet/SendingToAndFromManyAddressesSpecification_Steps.cs

```diff
@@ -0,0 +1,189 @@
+            TestHelper.WaitLoop(() => this.nodes[NodeTwo].CreateRPCClient().GetRawMempool().Length > 0);
+            TestHelper.WaitLoop(() => this.nodes[NodeTwo].FullNode.WalletManager().GetSpendableTransactionsInWallet(WalletName).Any());
+        }
+
+        private void node2_receives_the_funds()
+        {
+            this.stratisReceiverTotalAmount = this.nodes[NodeTwo].FullNode.WalletManager().
+                GetSpendableTransactionsInWallet(WalletName).
+                Sum(s => s.Transaction.Amount);
+
+            this.stratisReceiverTotalAmount.Should().Be(5000000000);
+
+            this.nodes[NodeTwo].FullNode.WalletManager().
+                GetSpendableTransactionsInWallet(WalletName).
+                First().Transaction.BlockHeight.
+                Should().BeNull();
+
+            this.nodes[NodeTwo].FullNode.WalletManager().
+                GetSpendableTransactionsInWallet(WalletName).
+                Count().Should().Be(50);
+
+            this.nodes[NodeOne].GenerateStratisWithMiner(1);
+
+            this.sharedSteps.WaitForBlockStoreToSync(this.nodes[NodeOne]);
+            this.sharedSteps.WaitForBlockStoreToSync(this.nodes[NodeTwo], this.nodes[NodeOne]);
+
+            // Confirm transactions have been committed to the block.
+            this.nodes[NodeTwo].FullNode.WalletManager().
+                GetSpendableTransactionsInWallet(WalletName).
+                First().Transaction.BlockHeight.
+                Should().Be(this.CoinBaseMaturity + 51);
+        }
+
+        private void node2_sends_funds_to_node1_FROM_fifty_addresses()
+        {
+            // Send coins to the node one.
+            HdAddress sendto = this.nodes[NodeOne].FullNode.WalletManager()
+                .GetUnusedAddress(new WalletAccountReference(WalletName, WalletAccountName));
+
+            this.transactionBuildContext = new TransactionBuildContext(
+                new WalletAccountReference(WalletName, WalletAccountName),
+                new[]
+                {
+                    new Recipient
+                    {
+                        Amount = this.stratisReceiverTotalAmount - Money.COIN,
+                        ScriptPubKey = sendto.ScriptPubKey
+                    }
+                }.ToList(), WalletPassword);
+
+            // Check node two has the correct inputs.
+            Transaction transaction = this.nodes[NodeTwo].FullNode.WalletTransactionHandler()
+                        .BuildTransaction(this.transactionBuildContext);
+
+            transaction.Inputs.Count.Should().Be(50);
+
+            // Broadcast.
+            this.nodes[NodeTwo].FullNode.NodeService<WalletController>()
+                .SendTransaction(new SendTransactionRequest(transaction.ToHex()));
+
+            // Wait for the transaction to arrive.
+            TestHelper.WaitLoop(() => this.nodes[NodeOne].FullNode.WalletManager()
+                                        .GetSpendableTransactionsInWallet(WalletName).Any());
```

Comment:
Is this check correct?  Doesn't it have some spendable transactions already before it receives this lot?

Reply:
probably was zero I guess  - perhaps we check that first before making it, to make sure this check is definitely correct?

Reply:
This was zero.  So the Any() check is good.